report.py

```diff
@@ -161,15 +161,72 @@ def run_report(config, blockers, server, header, test_email, no_email, template_
 	summary['total_unstable'] = "Total UNSTABLE: {}/{} = {}%".format(num_unstable, num_jobs, percent(num_unstable, num_jobs))
 	summary['total_failure'] = "Total FAILURE:  {}/{} = {}%".format(num_failure, num_jobs, percent(num_failure, num_jobs))
 
+	# Map job color codes to job count
+	jobs_dict = {
+		'#3465a4': num_success,
+		'#515151': num_aborted,
+		'#ef2929': num_failure,
+		'#704426': num_error,
+		'#ffb738': num_unstable,
+		'#bbbbbb': num_missing
+	}
+	# Map job type to job count
+	labels_dict = {
+		'Success': num_success,
+		'Aborted': num_aborted,
+		'Failed': num_failure,
+		'Error': num_error,
+		'Unstable': num_unstable,
+		'Missing': num_missing
+	}
```

Comment:
um, this looks a bit redundant, we can have the the values as tuples in form, (num_success, 'Success')

Reply:
Haven't tested it, but here it is
```
        jobs_dict = {'blue': (num_success, 'Success'), 'black': (num_aborted, 'Aborted'), 'red': (num_failure, 'Failure'), 'brown': (num_error, 'Error'), 'yellow': (num_unstable, 'Unstable'), 'grey': (num_missing, 'Missing')}
        bg_color_list, data_list, labels_list = [], [], []
        for key, value in jobs_dict.items():
            if value[0] != 0:
                bg_color_list.append(key)
                data_list.append(value[0])
                labels_list.append(value[1])

Reply:
@amodi5 : I tried it. Python tuples are immutable, so we cannot use append() to match job count data inside the for loop.

If we do not add this additional label to job type count, then see the below image and observe how the legends colors are mismatched for unstable and aborted jobs
![Screenshot from 2020-10-12 16-45-16](https://user-images.githubusercontent.com/46899831/95788920-7ad9cd80-0caa-11eb-8db8-69ee06e480a7.png)

Reply:
> @amodi5 : I tried it. Python tuples are immutable, so we cannot use append() to match job count data inside the for loop.
> 
> If we do not add this additional label to job type count, then see the below image and observe how the legends colors are mismatched for unstable and aborted jobs
> ![Screenshot from 2020-10-12 16-45-16](https://user-images.githubusercontent.com/46899831/95788920-7ad9cd80-0caa-11eb-8db8-69ee06e480a7.png)

Can you try with the following code, it worked for me
![unnamed (5)](https://user-images.githubusercontent.com/32521570/95789257-3438a300-0cab-11eb-9460-c5f2e5602590.png)
```
        jobs_dict = {'blue': (num_success, 'Success'), 'black': (num_aborted, 'Aborted'), 'red': (num_failure, 'Failure'), 'brown': (num_error, 'Error'), 'yellow': (num_unstable, 'Unstable'), 'grey': (num_missing, 'Missing')}
        bg_color_list, data_list, labels_list = [], [], []
        for key, value in jobs_dict.items():
            if value[0] != 0:
                bg_color_list.append(key)
                data_list.append(value[0])
                labels_list.append(value[1])